app/js/checkAnswers.js

```diff
@@ -1,28 +1,33 @@
 const questionAmount = 30// amount of questions
 
-document.querySelector('#finish').addEventListener('click', function(e) {
-    var unanswered = questionAnswered()
-    if (unanswered == false) {
+document.querySelector('#finish').addEventListener('click', function() {
+    let unanswered = questionAnswered()
+    if (unanswered == false && !Object.values(flaggedQuestions).includes(true)) {
         showResults()
-    } else {
-        document.getElementById('modal-title').textContent = `You have ${unanswered.length} unanswered questions.`
+    }
+    else {
+        let flagList = document.getElementById('flag-list')
+        let unansweredList = document.getElementById('unanswered-list')
+        document.getElementById('modal-title').textContent = 'You have ' + unanswered.length + ' unanswered questions.'
+        flagList.innerHTML = ''
+        unansweredList.innerHTML = ''
+        Object.values(flaggedQuestions).forEach(function(isQuestionFlagged, qId) {
+            if (isQuestionFlagged) {
+                flagList.innerHTML += "<li>" + (qId + 1) +"</li>"
+            }
+        })
+        unanswered.forEach(function (isQuestionUnanswered) {
+            unansweredList.innerHTML += "<li>" + isQuestionUnanswered +"</li>"
+        })
         openDialog()
-        document.querySelector('#modal-close').addEventListener('click', function() {
-            document.querySelector('#modal-finish').removeEventListener('click', finishTest)
+        document.querySelector('#modal-close').addEventListener('click', closeDialog)
+        document.querySelector('#modal-finish').addEventListener('click', function() {
+            showResults()
             closeDialog()
         })
-        document.querySelector('#modal-finish').addEventListener('click', finishTest)
```

Comment:
removing this and using an anonymous function means you send results twice when you click finish

Reply:
Changed in 068daec